backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java

```diff
@@ -351,13 +357,18 @@ public RepositoryEntity getProtectedRepositoryById(
   private Optional<RepositoryEntity> getRepositoryByName(
       Session session, String name, Workspace workspace, RepositoryTypeEnum repositoryType) {
     List<Long> repositoryIds = getRepositoryEntityIdsByName(session, name, repositoryType);
+    ModelDBServiceResourceTypes modelDBServiceResourceTypes =
+        ModelDBServiceResourceTypes.REPOSITORY;
+    if (repositoryType.equals(RepositoryTypeEnum.DATASET)) {
+      modelDBServiceResourceTypes = ModelDBServiceResourceTypes.DATASET;
+    }
```

Comment:
We should replace this with the helper function, but not sure how to do it right now.
```suggestion
    // TODO: replace with the helper function
    ModelDBServiceResourceTypes modelDBServiceResourceTypes =
        ModelDBServiceResourceTypes.REPOSITORY;
    if (repositoryType.equals(RepositoryTypeEnum.DATASET)) {
      modelDBServiceResourceTypes = ModelDBServiceResourceTypes.DATASET;
    }
```